dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/fat/AcmeValidityAndRenewTest.java

```diff
@@ -0,0 +1,355 @@
+public class AcmeValidityAndRenewTest {
+
+	@Server("com.ibm.ws.security.acme.fat.simple")
+	public static LibertyServer server;
+
+	private static ServerConfiguration ORIGINAL_CONFIG;
+	private static final String[] DOMAINS1 = { "domain1.com" };
+
+	public static CAContainer caContainer;
+	
+	public static final long timeBufferToExpire = 30000L; // milliseconds
+
+	@BeforeClass
+	public static void beforeClass() throws Exception {
+		ORIGINAL_CONFIG = server.getServerConfiguration();
+		try {
+			caContainer = new com.ibm.ws.security.acme.docker.pebble.PebbleContainer();
+		} catch (Exception e) {
+			caContainer = new com.ibm.ws.security.acme.docker.pebble.PebbleContainer();
+		}
+		AcmeFatUtils.checkPortOpen(caContainer.getHttpPort(), 60000);
+	}
+
+	@AfterClass
+	public static void afterClass() throws Exception {
+		if (caContainer != null) {
+			caContainer.stop();
+		}
+	}
+
+	@After
+	public void afterTest() {
+		/*
+		 * Cleanup any generated ACME files.
+		 */
+		AcmeFatUtils.deleteAcmeFiles(server);
+
+	}
+
+	/**
+	 * Whether to use "acme://" style URIs, or "https://" style URIs.
+	 * 
+	 * <p/>
+	 * Sub-classes can override this method to determine which style URIs should be
+	 * used.
+	 * 
+	 * @return Whether to use "acme://" style URIs.
+	 */
+	protected boolean useAcmeURIs() {
+		return false;
+	}
+
+	/**
+	 * This test will verify that the server starts up and requests a new
+	 * certificate from the ACME server when required.
+	 * 
+	 * @throws Exception If the test failed for some reason.
+	 */
+	@Test
+	@CheckForLeakedPasswords(AcmeFatUtils.PEBBLE_TRUSTSTORE_PASSWORD)
+	public void serverBelowMinRenew() throws Exception {
+		final String methodName = "serverBelowMinRenew";
+		Certificate[] startingCertificateChain = null, endingCertificateChain = null;
+
+		/*
+		 * Configure the acmeCA-2.0 feature.
+		 * 
+		 */
+
+		ServerConfiguration configuration = ORIGINAL_CONFIG.clone();
+		configuration.getAcmeCA().setRenewBeforeExpiration(AcmeConstants.RENEW_CERT_MIN - 1000 + "ms");
+		AcmeFatUtils.configureAcmeCA(server, caContainer, configuration, useAcmeURIs(), DOMAINS1);
+
+		/***********************************************************************
+		 * 
+		 * TEST 1: The server will start up without the specified keystore available. It
+		 * should generate a new keystore with a certificate retrieved from the ACME CA
+		 * server.
+		 * 
+		 **********************************************************************/
+		try {
+			Log.info(this.getClass(), methodName, "TEST renew is set to below the minimum renew time.");
+
+			/*
+			 * Start the server and wait for the certificate to be installed.
+			 */
+			server.startServer();
+			AcmeFatUtils.waitForAcmeToCreateCertificate(server);
+			AcmeFatUtils.waitForSslEndpoint(server);
+
+			/*
+			 * Verify that the server is now using a certificate signed by the CA.
+			 */
+			endingCertificateChain = AcmeFatUtils.assertAndGetServerCertificate(server, caContainer);
```

Comment:
probably don't care about this in this test, but i guess it doesn't hurt.

Reply:
Updated